src/toil/common.py

```diff
@@ -1511,20 +1514,29 @@ def parseSetEnv(l):
     return d
 
 
-def iC(minValue, maxValue=SYS_MAX_SIZE):
-    # Returns function that checks if a given int is in the given half-open interval
-    assert isinstance(minValue, int) and isinstance(maxValue, int)
-    return lambda x: minValue <= x < maxValue
+def iC(minValue: int, maxValue: int = SYS_MAX_SIZE) -> Callable[[int], bool]:
+    """Return function that checks if a given int is in the given half-open interval."""
+    if not (isinstance(minValue, int) and isinstance(maxValue, int)):
+        raise AssertionError
+
+    def func(value: int) -> bool:
+        return minValue <= value < maxValue
+    return func
 
 
-def fC(minValue, maxValue=None):
-    # Returns function that checks if a given float is in the given half-open interval
-    assert isinstance(minValue, float)
+def fC(minValue: float, maxValue: Optional[float] = None) -> Callable[[float], bool]:
+    """Return function that checks if a given float is in the given half-open interval."""
+    if not isinstance(minValue, float):
+        raise AssertionError
     if maxValue is None:
-        return lambda x: minValue <= x
+        def func(value: float) -> bool:
+            return minValue <= value
     else:
-        assert isinstance(maxValue, float)
-        return lambda x: minValue <= x < maxValue
+        def func(value: float) -> bool:
+            if not isinstance(maxValue, float):
+                raise AssertionError
```

Comment:
This is going to defer the check on maxValue until the returned function is run, so if you call `fC` with an unacceptable `maxValue` now you won't find out until later. I would leave the check in `fC` itself.